qiskit/execute.py

```diff
@@ -242,4 +245,8 @@ def execute(experiments, backend,
                     )
 
     # executing the circuits on the backend and returning the job
-    return backend.run(qobj, **run_config)
+    if type(backend).__name__ == 'IBMQBackend':
```

Comment:
I think `isinstance` would be better here.

Reply:
But then I need to import the instance to check this and Terra is supposed to be independent of the IBM Q provider.